contracts/apps/20-transfer/ICS20TransferBank.sol

```diff
@@ -18,29 +19,23 @@ contract ICS20TransferBank is ICS20Transfer {
 
     function sendTransfer(
         string calldata denom,
-        uint64 amount,
-        address receiver,
+        uint256 amount,
+        string calldata receiver,
         string calldata sourcePort,
         string calldata sourceChannel,
         uint64 timeoutHeight
     ) external {
-        if (!denom.toSlice().startsWith(_makeDenomPrefix(sourcePort, sourceChannel))) {
+        require(ICS20Lib.isEscapedJSONString(denom) && ICS20Lib.isEscapedJSONString(receiver), "unescaped string");
```

Comment:
i doubt you should validate. it will just burn gas. people already send right strings, or get pack fail ACK.

Reply:
First, `denom` can be removed by making this a contract to `IICS20Bank` implementers. I will revert a change in https://github.com/hyperledger-labs/yui-ibc-solidity/pull/213/commits/201e8f5a1646777f02c9c8af3d44d17b14bd338f

A format of `receiver` is not defined in ics-20, so I think this validation is basically necessary as it depends on the ics-20 impl on the counterparty chain.